android/src/main/java/com/usabilla/reactnative/ubform/UsabillaBridge.java

```diff
@@ -87,6 +87,27 @@ public void onReceive(Context context, Intent intent) {
         }
     };
 
+    /**
+     * Called via the index.js to handle back press
+     */
+    @ReactMethod
+    public void onBackPressed() {
+        final Activity activity = getCurrentActivity();
+        if (activity instanceof FragmentActivity) {
+            FragmentManager supportFragmentManager = ((FragmentActivity) activity).getSupportFragmentManager();
+            Fragment fragment = supportFragmentManager.findFragmentByTag(FRAGMENT_TAG);
+
+            if (fragment != null) {
+                supportFragmentManager.beginTransaction().remove(fragment).commit();
+            }
+            final WritableMap result = Arguments.createMap();
```

Comment:
Probably all this piece should go inside the `if` statement from before, since if there is no fragment then no action based on the form closing should be taken

Reply:
But I think, we need to fire `UBFormDidClose` event as we do in the receiver.

Reply:
Do we? From my understanding this is a way to handle the back button click to make it remove the form if present; which to me sounds that if the form is NOT present (aka the fragment is not there) nothing should happen